handler.go

```diff
@@ -46,6 +46,10 @@ type Spec struct {
 	// Vulcan middlewares to register with the handler. When registering, middlewares are assigned priorities
 	// according to their positions in the list: a middleware that appears in the list earlier is executed first.
 	Middlewares []middleware.Middleware
+
+	// When Handler or HandlerWith body is used, this function will be called after every request with a log message.
+	// If nil, defaults to github.com/mailgun/log.Infof.
+	Logger func(format string, a ...interface{})
```

Comment:
Argument for keeping the `Logger` parameter as-is: External users might not want to use the mailgun/log package. I have also seen people mention that they like the fact that net/http package does not do logging by default, because it lets them log what they want to. This parameter is also easy to use as-is while providing more options to the client -- see my other PR.